### PR TITLE
Add docs on esc/iac dependency tracking

### PR DESCRIPTION
**Pull request by seanyeh**
### Proposed changes

Add tips on esc/iac stack update tracking.
Also updated some outdated docs to include esc projects

Main changes:
Screenshot 2024-11-18 at 6 06 41 PM
Screenshot 2024-11-18 at 6 06 47 PM

### CODE REVIEW

File: content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md [hunk 1]
```diff
@@ -12,16 +12,20 @@ menu:
 
 With support for Pulumi ESC built into the Pulumi CLI, you can expose an environment's settings and secrets to any or all of your Pulumi stacks, bypassing the need to define and maintain individual configuration settings or secrets "locally" in Pulumi config files. The optional `pulumiConfig` key enables this.
 
-The following example updates the `myorg/myapp-dev` environment by adding a `pulumiConfig` block. This block specifies the [Pulumi configuration](/docs/concepts/config/) settings to expose to the Pulumi stack at runtime:
+{{% notes type="tip" %}}
```
Comment (arunkumar611) [inline content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md:15]: Curious why we use type=tip here? I've not seen tip used before. We usually do type=info

File: content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md [hunk 2]
```diff
@@ -12,16 +12,20 @@ menu:
...
+The `pulumi` CLI (as of v3.139.0) now tracks ESC environments used in stack updates. You can now view which ESC environments were used in your updates, right in the Pulumi Cloud Console!
```
Comment (arunkumar611) [inline content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md:16]: Add link to Pulumi CLI

Comment (arunkumar611) [inline content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md:16]:
```suggestion
The `pulumi` CLI (as of v3.139.0) now tracks ESC environments used in stack updates. You can now view which ESC environments were used in your updates on the Stack Overview page within the Pulumi Cloud Console. 
```

File: content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md (hunk 1 above)
Comment (seanyeh) [reply content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md:15]: No particular reason, I can change it to info.

File: content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md [hunk 3]
```diff
@@ -12,16 +12,20 @@ menu:
 
 With support for Pulumi ESC built into the Pulumi CLI, you can expose an environment's settings and secrets to any or all of your Pulumi stacks, bypassing the need to define and maintain individual configuration settings or secrets "locally" in Pulumi config files. The optional `pulumiConfig` key enables this.
 
-The following example updates the `myorg/myapp-dev` environment by adding a `pulumiConfig` block. This block specifies the [Pulumi configuration](/docs/concepts/config/) settings to expose to the Pulumi stack at runtime:
+{{% notes type="info" %}}
+The [pulumi](/docs/iac/cli/) CLI (as of v3.139.0) now tracks ESC environments used in stack updates. You can now view which ESC environments were used in your updates on the Stack Overview page within the Pulumi Cloud Console.
```
Comment (dschaller) [inline content/docs/esc/integrations/infrastructure/pulumi-iac/_index.md:16]:
```suggestion
The [pulumi](/docs/iac/cli/) CLI (as of v3.139.0) now tracks ESC environments used in stack updates. You can view which ESC environments were used in your updates on the Stack Overview page within the Pulumi Cloud Console.
```

**Review by dschaller — APPROVED** [review]

**Pull request merged**